File: ProgramContest/pc2-9.1.6/samps/sumit.pl

#
# File:    sum1.pl
# Purpose: to sum the integers in the file sumit.dat
# Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
#
# Sat Oct 30 13:20:59 PDT 1999
#
# caveat - this is not nice code, copy at own risk ;)
#
# $Id: sumit.pl 326 2006-10-25 02:53:57Z laned $
#
#

	open (FP, "sumit.dat") || die "Could not open file sumit.dat\n";

	$sum = 0;

	while (<FP>)
	{
		chomp;
		$num = 0 + $_;
		$sum += $num if $num > 0;

	}
	print "The sum of the integers is $sum\n";

# eof

File: ProgramContest/pc2-9.1.6/samps/sumit.cpp

//
// File:    sumit.cpp
// Purpose: to sum the integers in the file sumit.dat
// Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
//
// caveat - this is not nice code, copy at own risk ;)
//
// $Id: sumit.cpp 1633 2008-11-05 03:29:46Z laned $
//
//

#include <fstream>
#include <iostream>
using namespace std;

main ()
{
	int num;
	int sum;
	ifstream filein("sumit.dat");

	if (filein.fail())
	{
		cerr << "Could not read from file sumit.dat " << endl;
	}
	else
	{

		sum = 0;

		while(!filein.eof())
		{
			filein >> num;

			if (num > 0)
			{
				sum += num;
			}
		}

		cout << "The sum of the integers is " << sum << endl;

	}
}

// eof 
